services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineDocumentEditorPlugin.java

```diff
@@ -102,39 +129,75 @@ public void initActivity(String uuid, String workspace, String activityId) throw
     Node node = wopiService.getNode(symlink.getSession().getWorkspace().getName(), symlink.getPath());
     if (node != null) {
       String link = getEditorLink(node);
+      if (link != null) {
+        link = new StringBuilder("'").append(link).append("'").toString();
+      } else {
+        link = "null";
+      }
       callModule("officeonline.initActivity('" + node.getUUID() + "', " + link + ", '" + activityId + "');");
     }
   }
 
   /**
    * Inits the preview.
    *
-   * @param uuid the uuid
+   * @param <T> the generic type
+   * @param fileId the uuid
    * @param workspace the workspace
-   * @param activityId the activity id
-   * @param context the context
-   * @param index the index
-   * @throws Exception the exception
+   * @param requestURI the requestURI
+   * @param locale the locale
+   * @return the editor settings
    */
+  @SuppressWarnings("unchecked")
   @Override
-  public void initPreview(String uuid, String workspace, String activityId, String context, int index) throws Exception {
-    Node symlink = wopiService.nodeByUUID(uuid, workspace);
-    Node node = wopiService.getNode(symlink.getSession().getWorkspace().getName(), symlink.getPath());
-    if (node != null) {
-      if (symlink.isNodeType("exo:symlink")) {
-        String userId = WebuiRequestContext.getCurrentInstance().getRemoteUser();
-        wopiService.addFilePreferences(node, userId, symlink.getPath());
+  public EditorSetting initPreview(String fileId, String workspace, URI requestURI, Locale locale) {
+    try {
+      String userId = ConversationState.getCurrent().getIdentity().getUserId();
```

Comment:
Aquire this only where you actually consume it, below at line 160.

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/30/commits/2495d2de3e4c5cd7f2fa6dd97275a06344d55e73